Calendar/Get-CalendarDiagnosticObjectsSummary.ps1

```diff
@@ -0,0 +1,922 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+# .DESCRIPTION
+# This script runs the Get-CalendarDiagnosticObjects script and returns a summarized timeline of actions in clear english
+#
+# .PARAMETER Identity
+# Address of User Mailbox to query
+#
+# .PARAMETER Subject
+# Subject of the meeting to query
+#
+# .PARAMETER MeetingID
+# The MeetingID of the meeting to query
+#
+# .EXAMPLE
+# Get-CalendarDiagnosticObjectsSummary.ps1 -Identity someuser@microsoft.com -MeetingID 0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901
+#
+# Get-CalendarDiagnosticObjectsSummary.ps1 -Identity someuser@microsoft.com -Subject Test_OneTime_Meeting_Subject
+#
+#
+
+[CmdLetBinding()]
+param(
+    [Parameter(Mandatory)]
+    [string] $Identity,
+    [string] $Subject,
+    [string] $MeetingID
+)
+
+function GetCalendarDiagnosticObjects {
+    $CustomPropertyNameList = "AppointmentCounterProposal", "AppointmentRecurring", "CalendarItemType", "CalendarProcessed", "ClientIntent", "DisplayAttendeesCc", "DisplayAttendeesTo", "EventEmailReminderTimer", "ExternalSharingMasterId", "FreeBusyStatus", "From", "HasAttachment", "IsAllDayEvent", "IsCancelled", "IsMeeting", "MapiEndTime", "MapiStartTime", "OnlineMeetingConfLink", "OnlineMeetingExternalLink", "OnlineMeetingInternalLink", "SentRepresentingDisplayName", "SentRepresentingEmailAddress";
+    if ($Identity -and $Subject -and $MeetingID) {
+        $script:GetCDO = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults 2000;
+    }
+
+    if ($Identity -and $Subject -and !$MeetingID) {
+        $script:GetCDO = Get-CalendarDiagnosticObjects -Identity $Identity -Subject $Subject -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults 2000;
+    }
+
+    if ($Identity -and $MeetingID -and !$Subject) {
+        $script:GetCDO = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults 2000;
+    }
+
+    if ($Identity -and !$Subject -and !$MeetingID) {
+        Write-Warning "Can't run command with just Identity, Subject or MeetingID is also needed";
+        exit;
+    }
+}
+
+GetCalendarDiagnosticObjects;
```

Comment:
This should move down towards the bottom of script, so it isn't in the middle of a bunch of function declares and hidden.

Reply:
Moved down after last function.